app/src/main/java/com/kickstarter/ui/fragments/PledgeFragment.kt

```diff
@@ -486,6 +491,23 @@ class PledgeFragment : BaseFragment<PledgeFragmentViewModel.ViewModel>(), Reward
         return offsetViewBounds.top - parent.paddingTop
     }
 
+    private fun setDeadlineWarning(totalAndDeadline: Pair<String, String>) {
+        val total = totalAndDeadline.first
+        val deadline = totalAndDeadline.second
+        val ksString = this.viewModel.environment.ksString()
+        val warning = ksString.format(getString(R.string.If_the_project_reaches_its_funding_goal_you_will_be_charged_total_on_project_deadline),
+                "total", total,
+                "project_deadline", deadline)
+
+        val spannableWarning = SpannableString(warning)
+
+        ViewUtils.addBoldSpan(spannableWarning, total)
+        ViewUtils.addBoldSpan(spannableWarning, deadline)
+
+        deadline_warning.text = spannableWarning
+        deadline_warning.visibility = View.VISIBLE
```

Comment:
I wondering if setting the visibility of this view should be it's own view model output? Or alternatively we could consider re-naming the output from `totalAndDeadline` to something that also communicates that the label will also be set to `Visible` 🤔

Reply:
it should always be visible, i just set it to `visible` so it fades in because it depends on the total loading (which depends on the shipping rules call)